.github/workflows/ci.yml

```
@@ -62,25 +41,6 @@ jobs:
         GITHUB_WEBHOOK_SECRET: 'test-secret'
         GITHUB_TOKEN: 'test-token'
 
-  # Integration tests - moderate complexity
-  test-integration:
-    name: Integration Tests
-    runs-on: ubuntu-latest
-    
-    steps:
-    - name: Checkout code
-      uses: actions/checkout@v4
-
-    - name: Setup Node.js
-      uses: actions/setup-node@v4
-      with:
-        node-version: ${{ env.NODE_VERSION }}
-        cache: 'npm'
-        cache-dependency-path: 'package-lock.json'
-
-    - name: Install dependencies
-      run: npm ci --prefer-offline --no-audit
-
     - name: Run integration tests
       run: npm run test:integration || echo "No integration tests found, skipping"
```

The integration tests step contains fallback logic `|| echo "No integration tests found, skipping"` which could mask failures. Since this is the main CI pipeline, integration tests should be mandatory or the step should be more explicit about when it's okay to skip them.